ekd_distill.py

```
@@ -472,6 +483,20 @@ def main():
+    # Optional: for entropy-top-k-with-softmax, subsample to 1/20 of the dataset size
+    if getattr(config, "distill_type", "") == "entropy-top-k-with-softmax":
+        try:
+            from torch.utils.data import Subset
+            total_len = len(dataset)  # works for list or dataset-like
+            sub_len = max(1, total_len // 10)
+            if isinstance(dataset, list):
+                dataset = dataset[:sub_len]
+            else:
+                dataset = Subset(dataset, list(range(sub_len)))
+            print(f"[entropy-top-k-with-softmax] Using a 1/10 subset: {sub_len} / {total_len} examples")
+        except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)